dashboard_website/index.js

```diff
@@ -0,0 +1,419 @@
+        updateRange('10')
+        updateRangeLower('2')
+        drawBoxPlot();
+    });
+
+    // Add event listener to SNR select
+    const snrSelect = document.getElementById('snr-select');
+    snrSelect.addEventListener('change', function(event) {
+        selectedSNR = event.target.value;
+        updateRange('10')
+        updateRangeLower('2')
+        drawBoxPlot();
+    });
+
+    // Add event listener to type select
+    const typeSelect = document.getElementById('type-select');
+    typeSelect.addEventListener('change', function(event) {
+        selectedType = event.target.value;
+        updateRange('10')
+        updateRangeLower('2')
+        drawBoxPlot();
+    });
+
+    // Add event listeners to range slider and buttons
+    const rangeSlider = document.getElementById('range-slider');
+    const rangeValue = document.getElementById('range-value');
+    const decrementRange = document.getElementById('decrement-range');
+    const incrementRange = document.getElementById('increment-range');
+    function updateRange(value) {
+        selectedRange = value;
+        //rangeValue.textContent = value;
+        rangeSlider.value = value
+        drawBoxPlot();
+    }
+
+    rangeSlider.addEventListener('input', function(event) {
+        updateRange(event.target.value);
+    });
+
+    decrementRange.addEventListener('click', function() {
+        let newValue = parseInt(rangeSlider.value) - 2;
+        if (newValue >= 2) {
+            rangeSlider.value = newValue;
+            updateRange(newValue);
+        }
+    });
+
+    incrementRange.addEventListener('click', function() {
+        let newValue = parseInt(rangeSlider.value) + 2;
+        if (newValue <= rangeSlider.max) {
+            rangeSlider.value = newValue;
+            updateRange(newValue);
+        }
+    });
+
+    // Add event listeners to range slider and buttons
+    const rangeSliderLower = document.getElementById('lower-range-slider');
+    const rangeValueLower = document.getElementById('lower-range-value');
+    const decrementRangeLower = document.getElementById('decrement-lower-range');
+    const incrementRangeLower = document.getElementById('increment-lower-range');
+    function updateRangeLower(value) {
+        selectedRangeLower = value;
+        //rangeValue.textContent = value;
+        rangeSliderLower.value = value
+        drawBoxPlot();
+    }
+
+    rangeSliderLower.addEventListener('input', function(event) {
+        updateRangeLower(event.target.value);
+    });
+
+    decrementRangeLower.addEventListener('click', function() {
+        let newValue = parseInt(rangeSlider.value) - 2;
+        if (newValue >= 2) {
+            rangeSliderLower.value = newValue;
+            updateRangeLower(newValue);
+        }
+    });
+
+    incrementRangeLower.addEventListener('click', function() {
+        let newValue = parseInt(rangeSliderLower.value) + 2;
+        if (newValue <= rangeSliderLower.max) {
```

Comment:
Sometimes this prevents the increment from getting to the maximum value. If the click pushes it above the max it does nothing. It should just set it to the max.